src/ProxyClient/Symfony.php

```diff
@@ -52,10 +53,26 @@ public function refresh($url, array $headers = [])
     protected function configureOptions()
     {
         $resolver = parent::configureOptions();
-        $resolver->setDefaults([
-            'purge_method' => PurgeListener::DEFAULT_PURGE_METHOD,
-        ]);
+        $resolver->setDefault('purge_method', PurgeListener::DEFAULT_PURGE_METHOD);
+        $resolver->setAllowedTypes('purge_method', 'string');
+        $resolver->setDefault('purge_tags_header', PurgeListener::DEFAULT_PURGE_TAGS_HEADER);
+        $resolver->setAllowedTypes('purge_tags_header', 'string');
 
         return $resolver;
     }
+
+    /**
+     * Remove/Expire cache objects based on cache tags.
+     *
+     * @param array $tags Tags that should be removed/expired from the cache
+     *
+     * @return $this
+     */
+    public function invalidateTags(array $tags)
+    {
+        // TODO: how should we escape best here?
```

Comment:
I would like to escape `,` and maybe also `new line` but we need to unescape again in the `PurgeListener` so I'm not sure if using the same method as `Varnish` does is a good idea. Especially because `Varnish` just escapes by replacing both with a `_` which makes it impossible to unescape again. To be honest, I find tags containing a `,` strange anyway so I'd prohibit this in the `ResponseTagger` upfront so it cannot happen at all but I guess that would be a BC break. Wdyt?

Reply:
as far as i remember the logic with escaping was that we don't care about unescaping. the worst case would be that foo,bar and foo_bar collide, which would mean we invalidate too many things. its an edge case and one we decided we can live with, as its semantically correct (invalidating more then necessary is ok, the opposite would be a problem)

this works as long as escaping is done both when setting the tags and when creating invalidation requests.

Reply:
I see. So I'll just escape and don't care about unescaping, right? It's an edge case anyway, I really don't know why I would ever use a comma in a tag.
We could, however, also use the same header multiple times. There's nothing wrong in using

```
X-Cache-Tags: foobar
X-Cache-Tags: other tag
```

But I don't know if http-plug supports this?

Reply:
i'd replace all `,` with `_` but then collapse the tags into one header with , separation. in the listener, you can then still split by `,`.

if i remember correctly, the varnish client even splits this into multiple requests if the list of tags is getting too long, to avoid problems with too long headers.

Reply:
Implemented in b45271066225312c3752f05c574fefa20855382b. Sorry, I'm too used to phpunit, I never used mockery so it was way faster for me to use the "native" mock stuff for consecutive argument assertions :)

Reply:
Ah, and I implemented it so it's a more general solution. I thought that might be needed by other proxies too in the future. I did not adjust Varnish though because there's some regex stuff going on there and I figured that's maybe too specific.